server/v2/stf/sims_entry.go

```diff
@@ -0,0 +1,34 @@
+package stf
+
+import (
+	"context"
+	"iter"
+
+	"cosmossdk.io/core/header"
+	"cosmossdk.io/core/server"
+	"cosmossdk.io/core/store"
+	"cosmossdk.io/core/transaction"
+)
+
+// doSimsTXs constructs a function to simulate transactions in a block execution context using the provided simsBuilder.
+func (s STF[T]) doSimsTXs(simsBuilder func(ctx context.Context) iter.Seq[T]) doInBlockDeliveryFn[T] {
+	return func(
+		exCtx context.Context,
+		_ []T,
+		newState store.WriterMap,
+		hi header.Info,
+	) ([]server.TxResult, error) {
+		const key = "sims.header.time"
+		simsCtx := context.WithValue(exCtx, key, hi.Time) //nolint: staticcheck // using string key to decouple
+		var results []server.TxResult
+		var i int32
+		for tx := range simsBuilder(simsCtx) {
```

Comment:
This calls the sims message factories that were setup in sims_runner.doMainLoop()